File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    const nums_set = new Set(nums)
    let longest = 0
    let len = 0
    for(let i of nums_set){
        if(!(nums_set.has(i-1))){
            len = 0
            while(nums_set.has(i + len)){
                len +=1
            }
        }
        longest = len>longest ? len : longest
    }
    return longest
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        set_nums = set(nums)
        longest_seq = 0
        for val in set_nums:
            if (val-1) not in set_nums:
                length = 0
                while(val+length in set_nums):
                    length +=1
                longest_seq= max(length,longest_seq)
        return longest_seq